File: mobile/android/Source/sensitive-data-cleartext/sensitive-data-cleartext-SharedPreferences/app/src/main/java/com/insecure/ostorlab/sharedpreferences/MainActivity.java
package com.insecure.ostorlab.sharedpreferences;

import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.preference.PreferenceManager;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    // Fake AWS token (sensitive data)
    private static final String FAKE_AWS_ACCESS_KEY = "AKIAT3G7X2Q9L1P8W0RZ";
    private static final String FAKE_AWS_SECRET_KEY = "wJalrXUtnFEMI/K7MDENG+bPxRfiCY7NQ3T9W5X4Z";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        storeSensitiveDataInSharedPreferences();
    }

    private void storeSensitiveDataInSharedPreferences() {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        // Intentionally storing sensitive data in plaintext
        editor.putString("aws_access_key_id", FAKE_AWS_ACCESS_KEY);
        editor.putString("aws_secret_access_key", FAKE_AWS_SECRET_KEY);
        editor.apply(); // or .commit()
    }
}

File: mobile/android/oxo-android-ben64/src/app/src/main/java/com/example/oxo_android_ben64/MainActivity.kt
package com.example.oxo_android_ben64

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.webkit.WebView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val myWebView: WebView = findViewById(R.id.webview)
        myWebView.settings.javaScriptEnabled = true

        myWebView.loadUrl("file:///android_asset/page.html?auth_token=abc123xyz&user_id=456")
    }
}
